src/net.rs

```diff
@@ -168,6 +168,9 @@ pub struct DkgPrivateBegin {
     pub signer_ids: Vec<u32>,
     /// Key IDs who responded in time for this DKG round
     pub key_ids: Vec<u32>,
+    /// Include DkgPublicShares to avoid p2p related message delivery
+    /// order issues when signers communicate directly with each other
+    pub dkg_public_shares: HashMap<u32, DkgPublicShares>,
```

Comment:
Can we make this a `BTreeMap`? Actually, maybe it's better to make this a more comprehensive change that gets done in another PR.

Reply:
Yeah we agree in the log comment thread above to settle the HashMap/BTreeMap debate in another PR.